src/Entity/BaseEntityManager.php

```diff
@@ -85,6 +88,28 @@ protected function getRepository(): EntityRepository
     {
         return $this->getEntityManager()->getRepository($this->class);
     }
+
+    public function beginTransaction(): void
+    {
+        $this->getEntityManager()->beginTransaction();
+    }
+
+    /**
+     * @throws TransactionException
+     */
+    public function commit(): void
+    {
+        try {
+            $this->getEntityManager()->commit();
+        } catch (Throwable $exception) {
+            throw new TransactionException($exception->getMessage(), $exception->getCode(), $exception);
```

Comment:
Hey @VincentLanglet 

do like this

or

```php
<?php

namespace Sonata\Doctrine\Exception;

use Exception;
use Throwable;

class TransactionException extends Exception
{
    public function __construct(Throwable $previous)
    {
        parent::__construct('Some message', 0, $previous);
    }
}
```

Reply:
```
public function __construct(Throwable $previous) { parent::__construct('Some message', 0, $previous); }
```
would be a bad idea because people wouldn't be able to create an exception (without another one) and it would force the error message.

I prefer to keep it this way.
People can then throw custom exception in their own implementation.